### DISCUSSION TITLE
ILI9341 reset pin held high problem

### DISCUSSION THREAD

**Discussion by adamalfath** [Bodmer/TFT_eSPI#2040] [Q&A - General] [answered]
Hi,

I'm designing my custom display board based on this [ILI9341 TFT Display](https://www.aliexpress.com/item/32884267326.html). Aiming to have minimum wire, I decided not to broken-out the RESET, CS & MISO pin like the schematic below:
![Screenshot_1](https://user-images.githubusercontent.com/19376004/192097459-a1c7e1d1-6185-4664-826f-c68856f6bf6e.png)
`RESET` pulled high to always enable the display, `CS` pulled low for always active SPI connection, and `MISO` left floating since I'm not using read command (?). No other SPI device is connected on this bus other than the display itself.

During testing, I've got issues when sometimes the display is responding to the command (firmware) and sometimes (most of the time) is not. Sometimes disconnecting the display, plugging it back, then hit MCU reset solve it, but it is not reproducible since right now that method is not working. Also my mistake of not having a "working" module to test around first before making my own -_-.

Here's what I've done:
1. Set `Setup70c_ESP32_C3_ILI9341.h` inside `User_Setup_Select.h` file
2. Define `TFT_MOSI=7`, `TFT_SCLK=6`, `TFT_DC=5` inside `Setup70c_ESP32_C3_ILI9341.h`, anything else is commented
3. Trying `ILI9341_DRIVER` or `ILI9341_2_DRIVER`
4. Test code:
    ```
    tft.init();
    delay(500);
    tft.fillScreen(TFT_RED);
    delay(500);
    tft.fillScreen(TFT_GREEN);
    delay(500);
    tft.fillScreen(TFT_BLUE);
    delay(500);
    tft.fillScreen(TFT_WHITE);
    delay(500);
    ```

Maybe someone can give me an insight whether this is hardware due to another necessary pin needed for communication or firmware (library config) issue. Any help is appreciated. Thank you.

**Comment by TangerineDreamer** [comment]
When RST goes to 3,3V, i suggest you write : 
 TFT_RST   -1

**Reply by adamalfath** [reply]
Hi, thanks for the response.

From my understanding, I think using TFT_RST -1 or commenting it would be the same as later it treated the same:
https://github.com/Bodmer/TFT_eSPI/blob/2b576dc614cbcd8f8e6cc583193efed40930d9c8/TFT_eSPI.cpp#L560-L565
https://github.com/Bodmer/TFT_eSPI/blob/2b576dc614cbcd8f8e6cc583193efed40930d9c8/TFT_eSPI.cpp#L673-L689

My problem is line [686](https://github.com/Bodmer/TFT_eSPI/blob/2b576dc614cbcd8f8e6cc583193efed40930d9c8/TFT_eSPI.cpp#L686)/[688 ](https://github.com/Bodmer/TFT_eSPI/blob/2b576dc614cbcd8f8e6cc583193efed40930d9c8/TFT_eSPI.cpp#L688)I mentioned above didn't have an affect to the display module. The module won't responding to the next command the MCU sent.

**Comment by adamalfath** [comment]
Update, after RTFM (duh), turns out my problem is clearly documented on the datasheet:
![Screenshot_2](https://user-images.githubusercontent.com/19376004/193424865-17da9546-e1f1-4638-8e88-6c261d946e8c.png)
The current workaround is to add delay to the reset signal by implementing RC circuit:
![Screenshot_3](https://user-images.githubusercontent.com/19376004/193425049-e7efaed0-10f6-473b-8f7b-bb045717966c.png)
The caveat is that this method only works during the first power-up since the reset pin is only triggered then. If I implement software reset to the main MCU (like `ESP.reset`/`ESP.restart`), the next initialization (during `init` function call) will make the display not respond to any command after.

**Comment by adamalfath** [comment, marked as answer]
This maybe became self-answered. I leave it here in case somebody needed it later.

|Solution|Note|
|:-|:-|
|Add delay circuit|Only works during first power-up|
|Tied display reset line to MCU reset line|Won't be responding to MCU soft reset|
|Dedicated HW reset signal from MCU IO|The best option|

**Comment by TangerineDreamer** [comment]
I find it such complicated to drive a tft 9341 with a esp32XX,have you tried with a simple esp32?
this setup works for me:

//--------------------------------------------------------------------------

//ILI9341 & ESP32

#define ILI9341_DRIVER




#define TFT_MISO 19 (better on the air)
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS   15  // Chip select control pin
#define TFT_DC    2  // Data Command control pin
//#define TFT_RST   4  // Reset pin (could connect to RST pin)
#define TFT_RST  -1  // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST

//#define TOUCH_CLK 18
#define TOUCH_CS 27 // Chip select pin (T_CS) of touch screen (if you don't have touch,comment it)
//#define TOUCH_DIN 23 
//#define TOUCH_DO 19 






#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:-.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
//#define LOAD_FONT8N // Font 8. Alternative to Font 8 above, slightly narrower, so 3 digits fit a 160 pixel TFT
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts


#define SMOOTH_FONT



#define SPI_FREQUENCY  20000000

#define SPI_READ_FREQUENCY  20000000


#define SPI_TOUCH_FREQUENCY  2500000

//-------------------------------------------------------------------------------

In this way you can define if your tft is alright,and maybe if there is bugs in the lastest esp32 packages,the 2.05 have some bugs,the
2.04 too,i had to downgrade to 2.02,that still support esp32sx series,i think.
